auth_mellon_util.c

```diff
@@ -126,7 +133,30 @@ int am_validate_redirect_url(request_rec *r, const char *url)
     }
 
     if (!uri.hostname) {
-        return OK; /* No hostname to check. */
+        /* No hostname to check. Instead, let's make sure that the path starts
+         * with a single "/" but not "//"
```

Comment:
Please xplain why this is important and how it fixes the CVE.
Otherwise it may get inadvertently removed/changed in ways that will re-introduce security issues.

Reply:
OK, taking @awakenine's comment into account, I should probably split the PR into two, one that is minimal and fixes the CVE and the other patches (and maybe tests even separately if we're going to have an issue with the license..)

Anyway, what the patch tries to do is to make sure that all relative redirects (those that do not contain an explicit host) must be prefixed with a slash to make sure the redirect is anchored at the root of the application and you can't "escape" elsewhere. This is not strictly related to the CVE, but I still think it is a good idea given that there already have been two or three CVEs caused by different parsing of different semi malformed URIs by the APR helper utilities and the browsers, when the URI is OK-ed by the module and then put into the Location header for the browser to process.